src/domain/orders/edit/modal.tsx

```diff
@@ -367,43 +365,41 @@ type OrderEditModalContainerProps = {
   order: Order
 }
 
+let isRequestRunningFlag = false
+
 function OrderEditModalContainer(props: OrderEditModalContainerProps) {
   const { order } = props
   const notification = useNotification()
 
-  const {
-    hideModal,
-    orderEdits,
-    activeOrderEditId,
-    setActiveOrderEdit,
-  } = useContext(OrderEditContext)
+  const { hideModal, orderEdits, activeOrderEditId, setActiveOrderEdit } =
+    useContext(OrderEditContext)
 
-  const { mutate: createOrderEdit } = useAdminCreateOrderEdit()
+  const { mutateAsync: createOrderEdit } = useAdminCreateOrderEdit()
 
   const orderEdit = orderEdits?.find((oe) => oe.id === activeOrderEditId)
 
   useEffect(() => {
-    if (activeOrderEditId) {
+    if (activeOrderEditId || isRequestRunningFlag) {
       return
     }
 
-    createOrderEdit(
-      { order_id: order.id },
-      {
-        onSuccess: ({ order_edit }) => {
-          setActiveOrderEdit(order_edit.id)
-        },
-        onError: () => {
-          notification(
-            "Error",
-            "There is already active order edit on this order",
-            "error"
-          )
-          hideModal()
-        },
+    isRequestRunningFlag = true
```

Comment:
Suggestion : instead of creating an async function that is immediately called. Why don’t we call the create and then .then .catch .finally (finally depending on the previous comment). Wdyt?